pkg/controller/direct/tags/tagbinding_controller.go

```diff
@@ -0,0 +1,197 @@
+	return &modelTagsTagBinding{config: *config}, nil
+}
+
+var _ directbase.Model = &modelTagsTagBinding{}
+
+type modelTagsTagBinding struct {
+	config config.ControllerConfig
+}
+
+func (m *modelTagsTagBinding) client(ctx context.Context, location string) (*resourcemanager.TagBindingsClient, error) {
+	var opts []option.ClientOption
+	opts, err := m.config.RESTClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	if location != "" {
+		opts = append(opts, option.WithEndpoint(fmt.Sprintf("%s-cloudresourcemanager.googleapis.com:443", location)))
+	}
+	gcpClient, err := resourcemanager.NewTagBindingsRESTClient(ctx, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("building TagsTagBinding client: %w", err)
+	}
+	return gcpClient, err
+}
+
+func (m *modelTagsTagBinding) AdapterForObject(ctx context.Context, reader client.Reader, u *unstructured.Unstructured) (directbase.Adapter, error) {
+	obj := &krm.TagsTagBinding{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &obj); err != nil {
+		return nil, fmt.Errorf("error converting to %T: %w", obj, err)
+	}
+
+	id, err := obj.GetIdentity(ctx, reader)
+	if err != nil {
+		return nil, err
+	}
+
+	var location string
+	if obj.Spec.Location != nil {
+		location = *obj.Spec.Location
+	}
+
+	gcpClient, err := m.client(ctx, location)
+	if err != nil {
+		return nil, fmt.Errorf("error creating gcp client: %w", err)
+	}
+	return &TagsTagBindingAdapter{
+		id:        id.(*krm.TagBindingIdentity),
+		gcpClient: gcpClient,
+	}, nil
+}
+
+func (m *modelTagsTagBinding) AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error) {
+	return nil, nil
+}
+
+type TagsTagBindingAdapter struct {
+	id        *krm.TagBindingIdentity
+	gcpClient *resourcemanager.TagBindingsClient
+	actual    *resourcemanagerpb.TagBinding
+}
+
+var _ directbase.Adapter = &TagsTagBindingAdapter{}
+
+func (a *TagsTagBindingAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting TagsTagBinding", "name", a.id)
+
+	req := &resourcemanagerpb.ListTagBindingsRequest{
+		Parent: a.id.Parent().String(),
+		// TODO: PageSize and PageToken
+	}
+	it := a.gcpClient.ListTagBindings(ctx, req)
+
+	for {
+		tagBinding, err := it.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			return false, err
+		}
+		if tagBinding.TagValue == a.id.TagValue() {
+			a.actual = tagBinding
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (a *TagsTagBindingAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating TagsTagBinding", "name", a.id)
+
+	req := &resourcemanagerpb.CreateTagBindingRequest{
+		TagBinding: &resourcemanagerpb.TagBinding{
+			Parent:   a.id.Parent().String(),
+			TagValue: a.id.TagValue(),
+		},
+	}
+	op, err := a.gcpClient.CreateTagBinding(ctx, req)
+
```

Comment:
Nit: don't put a line break here, if you want a line break put it after you handle `err`